src/utils/sources-tree/tests/getUrl.spec.js

```diff
@@ -0,0 +1,78 @@
+import { getURL } from "../getURL";
+import Url from "url";
+
+let spy;
+
+describe("getUrl", () => {
+  it("handles normal url with http and https for filename", function() {
+    const urlObject = getURL({ url: "https://a/b.js" });
+    expect(urlObject.filename).toBe("b.js");
+
+    const urlObject2 = getURL({ url: "http://a/b.js" });
+    expect(urlObject2.filename).toBe("b.js");
+  });
+
+  it("handles url with querystring for filename", function() {
+    const urlObject = getURL({ url: "https://a/b.js?key=randomeKey" });
```

Comment:
Don't kill me but there's an extra `e` in the `key` value...

Reply:
haha - love it